### PR TITLE
feat(Enzyme): add enzyme, tests cleanup

### PR DESCRIPTION
**Pull request by kosanna**
- Remove toJSON everywhere
- Update snapshots
- Tests cleanup

🥃 

Long term goal is decommissioning of `react-test-renderer`, `sinon` and `chai`

### CODE REVIEW

File: test/jest.config.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "testPathIgnorePatterns": [
-    "<rootDir>[/\\\\](dist|node_modules)[/\\\\]"
+    "<rootDir>/(dist|node_modules)/"
```
Comment (markdalgleish) [inline test/jest.config.json]: Why was this change made? IIRC this was for Windows support.

File: test/jest.config.json [hunk 2]
```diff
@@ -9,7 +9,7 @@
     "^.+\\.js$": "<rootDir>/test/babel.transform.js",
     "^.+\\.svg$": "<rootDir>/test/svg.transform.js"
   },
-  "transformIgnorePatterns": [
-    "[/\\\\]node_modules[/\\\\].+\\.(js|jsx)$"
```
Comment (markdalgleish) [inline test/jest.config.json]: Why was this removed? (Not saying it shouldn't have been, just curious)

**Comment by mengtzu** [comment]
This looks good, but I'm not sure I should approve commits made on a long weekend on general principle :p

**Comment by markdalgleish** [comment]
One concern I have here is that we've migrated some tests to use a shallow renderer. This definitely cleans up the snapshots, but it also means we're reducing test coverage in large components like the header and footer. Any thoughts on how we should deal with this? Should we add shallow snapshot tests for all the sub-components?

**Comment by mattsjones** [comment]
My feelings here are that the header and footer should be deep rendered snapshots. The components don't make sense bu themselves so they should be rendered as one.

**Comment by kosanna** [comment]
We're not loosing any actual tests with replacing to shallow anywhere in this case. If something need to be tests deeply, we need to test nested components themselves.

And btw, sinon could go later on too.

File: test/jest.config.json (hunk 2 above)
Comment (kosanna) [inline test/jest.config.json]: Because I didn't understand why it should be there and it works without it :P

File: test/jest.config.json (hunk 1 above)
Comment (kosanna) [inline test/jest.config.json]: Ok, got it, can roll it back

File: test/jest.config.json (hunk 2 above)
Comment (kosanna) [inline test/jest.config.json]: Ah, sorry, I thought you mean [/\\\\]

Comment (kosanna) [inline test/jest.config.json]: Had issues with toJson serialiser, looks like it's fine now

**Comment by markdalgleish** [comment]
We're not losing tests only in terms of the number of assertions, but we've lost test coverage in the sense that there are now changes that can be made to the header that won't get picked up in our test suite. I feel like this PR should either revert back to deep snapshots where appropriate, or introduce the missing snapshot tests.

**Comment by kosanna** [comment]
Comments are addressed @here. Header and Footer snapshots are fully rendered now. Please revisit the PR :)

**Comment by kosanna** [comment]
Anyone's keen for this PR to go in?

File: react/Footer/Footer.test.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { render } from 'enzyme';
```
Comment (markdalgleish) [inline react/Footer/Footer.test.js]: 👍
